packages/pagination/src/index.ts

```diff
@@ -215,3 +217,75 @@ export const createPagination = (
 
   return [paginationProps, page, setPage as Setter<number>];
 };
+
+declare module "solid-js" {
+  namespace JSX {
+    interface Directives {
+      infiniteScrollLoader: boolean;
+    }
+  }
+}
+
+export type _E = JSX.Element;
+
+/**
+ * Provides an easy way to implement infinite scrolling.
+ *
+ * ```ts
+ * const [pages, loader, { page, setPage, setPages, end, setEnd }] = createInfiniteScroll(fetcher);
+ * ```
+ * @param fetcher `(page: number) => Promise<T[]>`
+ * @return `pages()` is an accessor contains array of contents
+ * @property `pages.loading` is a boolean indicator for the loading state
+ * @property `pages.error` contains any error encountered
+ * @return `infiniteScrollLoader` is a directive used to set the loader element
+ * @method `page` is an accessor that contains page number
+ * @method `setPage` allows to manually change the page number
+ * @method `setPages` allows to manually change the contents of the page
+ * @method `end` is a boolean indicator for end of the page
+ * @method `setEnd` allows to manually change the end
+ */
+export function createInfiniteScroll<T>(fetcher: (page: number) => Promise<T[]>): [
+  pages: Accessor<T[]>,
+  loader: Directive,
+  options: {
+    page: Accessor<number>;
+    setPage: Setter<number>;
+    setPages: Setter<T[]>;
+    end: Accessor<boolean>;
+    setEnd: Setter<boolean>;
+  }
+] {
+  const [loader, setLoader] = createSignal<HTMLElement>();
+  const [pages, setPages] = createSignal<T[]>([]);
+  const [page, setPage] = createSignal(0);
+  const [end, setEnd] = createSignal(false);
+
+  const visible = createVisibilityObserver()(loader);
```

Comment:
Not sure if using `createVisibilityObserver` is necessary here. We don't care about having a `visible` signal, we just want to fire a callback when the element is in view — probably better to use `makeIntersectionObserver` to not needlessly create another signal. Unless I'm missing something.

Reply:
yes callback when element is in view. i just checked the package and used `createVisibilityObserver` because it was very simple.